tmt/steps/report/reportportal.py

```diff
@@ -39,112 +36,169 @@ class ReportReportPortalData(tmt.steps.report.ReportStepData):
+        assert self.step.plan.name is not None
+        launch_name = self.get("launch") or self.step.plan.name
+
+        version = "v1"
+        url = f"{server}/api/{version}/{project}"
+        headers = {"Authorization": "bearer " + token,
+                   "accept": "*/*",
+                   "Content-Type": "application/json"}
+
+        TMT_env_vars = self.get("env-vars") or []
+        attributes = [{'key': k, 'value': v[0]} for k, v in self.step.plan._fmf_context.items()]
+
+        # Communication with RP instance
         with tmt.utils.retry_session() as session:
-            url = f"{server}/api/v1/{project}/launch/import"
-            self.debug(f"Send the report to '{url}'.")
-            response = session.post(
-                url,
-                headers={
-                    "Authorization": "bearer " + token,
-                    "accept": "*/*",
-                    },
-                files={
-                    # The zip filename is used as the launch name in ReportPortal
-                    "file": (launch_name + ".zip", bytestream, "application/zip"),
-                    },
-                )
-
-        # Handle the response
-        try:
-            message = tmt.utils.yaml_to_dict(response.text).get("message")
-        except (tmt.utils.GeneralError, KeyError):
-            message = response.text
-        if not response.ok:
-            raise tmt.utils.ReportError(
-                f"Received non-ok status code from ReportPortal, response text is: {message}")
-        else:
-            self.debug(f"Response code from the server: {response.status_code}")
-            self.debug(f"Message from the server: {message}")
-            self.info("report", "Successfully uploaded.", "yellow")
+
+            # Create a launch
+            self.info("launch", launch_name, color="green")
+            response = session.post(url=f"{url}/launch",
+                                    headers=headers,
+                                    json={"name": launch_name,
+                                          "description": self.step.plan.summary,
+                                          "attributes": attributes,
+                                          "startTime": timestamp()})
+            self.handle_response(response)
+            launch_uuid = tmt.utils.yaml_to_dict(response.text).get("id")
+            assert launch_uuid is not None
+            self.info("    uuid", launch_uuid, "yellow")
+
+            # For each test
+            for result in self.step.plan.execute.results():
+                test = [test for test in self.step.plan.discover.tests()
+                        if test.serialnumber == result.serialnumber][0]
+                attributes.append({"key": "contact", "value": test.contact[0]})
+                env_vars = [{'key': k, 'value': v} for k, v in test.environment.items()
+                            if not k.startswith("TMT_") or k in TMT_env_vars]
```

Comment:
Assume I would like to add my 
self.get("env-vars")  Assume one would like to upload its own attributes, 

TMT_env_vars is not uploaded to attributes section

Reply:
It always uploads all know environment variables (under fmf data -> "environment"), except those that begin with "TMT_".
We decided to omit these TMT-variables as they tend to be unique per each run and break history aggregation.
If you want to display any of the TMT-variables, list them in optional argument env-vars, e.g. "--env-vars [TMT_TREE,TMT_PLAN_DATA]".

Reply:
Quick note: `--env-vars [foo, bar]` is a bit puzzling, the `[` and `]` have no meaning, IIUIC. In the code, `--env-vars` is retrieved like this:

```python
TMT_env_vars = self.get("env-vars") or []
```

When it's set by the user, `TMT_env_vars` is a **string**, otherwise it's an empty **list**. What you get is `"[TMT_TREE,TMT_PLAN_DATA]"`, not `["TMT_TREE", "TMT_PLAN_DATA"]` as you might have expected.

This is in conflict with the schema bit of the PR where `env-vars` is defined as an "array of strings" - when specified on the command line, it will not be a list of strings, but a string.

It would be better use `multiple=True` in `env_vars` field definition, and allow repetition of this option I.e. `--env-var TMT_TREE --env-var TMT_PLAN_DATA` - `self.get('env-var')` would then give you a list of strings, `['TMT_TREE', 'TMT_PLAN_DATA']`. This would be aligned with input from fmf files.

Reply:
Thank you, @happz.
Good point, I didn't know about this when I was trying to find a way how to get a list.
Although this way `"[TMT_TREE,TMT_PLAN_DATA]"`, it does work as well.

Reply:
Yep, it would work, `TMT_TREE in "[TMT_TREE, TMT_PLAN_DATA]"` would be true-ish, but I would argue we should avoid it and make things right :) The schema describes the field as a list of strings, it's shown as a list of strings in examples, it's even an empty list if not set, therefore keeping its string form around would create ambiguity for developers. One day, `for env_var in env_vars` might be needed, and our poor future selves would get `[`, `T`, `R`, ...

Reply:
Agreed. Working on update.
Although isn't there any tmt way to define it as list so it can be used as `--env-var TMT_TREE TMT_PLAN_DATA` and gives ['TMT_TREE', 'TMT_PLAN_DATA']

Reply:
Unfortunately, there's no out-of-the-box solution for this, it's either repeating the option, or manual parsing of the value is needed.

Reply:
Okay, now I found out it can be written without repeating in fmf data. Then it is no big deal to repeat it in CL only.